src/checkqueue.h

```diff
@@ -143,6 +164,14 @@ class CCheckQueue
         return Loop(true);
     }
 
+    //! Quit execution of any remaining checks.
+    void Quit()
+    {
+       boost::mutex::scoped_lock lock(mutex_fQuit);
+       fQuit = true;
+       LogPrint("parallel_2", "setting fQuit to: %d\n", fQuit);
```

Comment:
It doesn't makes sens to log while holding that mutex. Critical sections needs to be reduced to the minimum.